src/ucs/memory/memtype_cache.c

```diff
@@ -300,6 +315,30 @@ UCS_PROFILE_FUNC(ucs_status_t, ucs_memtype_cache_lookup,
     return status;
 }
 
+int ucs_memtype_cache_is_empty()
+{
+    return (ucs_memtype_cache_global_instance &&
+            !ucs_memtype_cache_global_instance->pgtable.num_regions);
+}
+
+static ucs_status_t ucs_memtype_cache_init_handler()
```

Comment:
ucs_memtype_cache_set_event_handler

Reply:
removed this function and moved back the logic back to the constructor based on your suggestion